src/hotspot/share/runtime/javaThread.hpp

```diff
@@ -81,9 +82,14 @@ class JavaThread: public Thread {
   friend class HandshakeState;
   friend class Continuation;
   friend class Threads;
+  friend class ServiceThread; // for deferred OopHandle release access
```

Comment:
Does this still have to be a friend?  You could make the two functions that ServiceThread calls public.

Reply:
I made them private and kept ServiceThread a friend so that it was more clear that this is a private interface with the ServiceThread and not some public API that might be called from other code.

Reply:
Thanks for doing it via a 'friend' and including a comment.